packages/typespec-python/src/emitter.ts

```diff
@@ -67,9 +67,23 @@ interface HttpServerParameter {
 const defaultOptions = {
     "package-version": "1.0.0b1",
     "generate-packaging-files": true,
-    "unbranded": false,
 };
 
+// case insensitive, known values are "azure" and others
+let flavor: string;
+
+function CalculateFlavor(options: PythonEmitterOptions & InternalPythonEmitterOptions): string {
+    if (options["flavor"] !== undefined) {
+        return options["flavor"].toLowerCase();
+    } else if (options["package-dir"] !== undefined && options["unbranded"] === undefined) {
+        if (options["package-dir"].toLowerCase().includes("azure")) {
+            return "azure";
+        }
+        return "unbranded";
+    }
+    return options["unbranded"] === false ? "azure" : "unbranded";
```

Comment:
I said yesterday to remove `unbranded`

Reply:
I understand you mean to remove `--unbranded`? I see.
nit: it is better to update the issue description https://github.com/Azure/autorest.python/issues/2429#issue-2164085654
![image](https://github.com/Azure/autorest.python/assets/70930885/76181875-abdc-4fd7-be39-8a1780f20bfe)

Reply:
FYI: I am not sure whether the word "unbranded" is still preferred as default value for `flavor`. So I change to "non-azure".